src/main/java/io/confluent/connect/jdbc/BulkTableQuerier.java

```diff
@@ -78,6 +78,15 @@ public SourceRecord extractRecord() throws SQLException {
       default:
         throw new ConnectException("Unexpected query mode: " + mode);
     }
+
+    if (keyColumn != null && keySchema != null) {
+      Object key = record.get(keyColumn);
+
+      if (key != null) {
```

Comment:
I think I was writing code defensively here in the case where the user specifies a key column but that column contains NULLs in the db for some reason. Obviously this seems unlikely but we need to make a decision on what to do. Should that be considered an error? I don't know if there is really a 'correct' behavior here, I guess it depends if we prefer the driver to be defensive or ensure all data gets loaded regardless of the configuration not making sense.